pyhap/characteristic.py

```diff
@@ -166,6 +153,23 @@ def get_value(self):
         """
         return self.value
 
+    def override_properties(self, properties=None, valid_values=None):
+        """Override characteristic property values and valid values.
+
+        :param properties: Dictionary with values to override the existing
+            properties. Only changed values are required.
+        :type properties: dict
+
+        :param valid_values: Dictionary with values to override the existing
+            valid_values. Valid values will be set to new dictionary.
+        :type valid_values: dict
+        """
+        if properties:
+            self.properties.update(properties)
+
+        if valid_values:
```

Comment:
I would add:
```
if valid_values:
    if not "ValidValues" in self.properties:
        raise ValueError(...)
    else:
        ...
```

Reply:
Actually forget the above - I will check if you can set ValidValues on chars that don't have ValidValues defined and will get back to you or will submit a change myself.